File: Examples/Python/open_and_ping.py
import helpers
import ximu3

# Create connection info
connection_info = ximu3.UsbConnectionInfo("COM1")

# Open and ping
connection = ximu3.Connection(connection_info)


def print_ping_response(ping_response):
    print(ping_response.interface + ", " +
          ping_response.device_name + " - " +
          ping_response.serial_number)
    # print(ping_response.to_string()) # alternative to above


def callback(result):
    if result != "Ok":
        print("Unable to open connection")
        return

    print_ping_response(connection.ping())


if helpers.yes_or_no("Use async implementation?"):
    connection.open_async(callback)
    helpers.wait(3)
else:
    connection.open()
    print_ping_response(connection.ping())

# Close connection
connection.close()

File: Examples/CSharp/x-IMU3-Examples/Examples/OpenAndPing.cs
using System;

namespace Ximu3Examples
{
    class OpenAndPing
    {
        public OpenAndPing()
        {
            // Create connection info
            Ximu3.UsbConnectionInfo connectionInfo = new Ximu3.UsbConnectionInfo("COM1");

            // Open and ping
            Ximu3.Connection connection = new Ximu3.Connection(connectionInfo);
            if (Helpers.YesOrNo("Use async implementation?"))
            {
                connection.OpenAsync(OpenEvent);
                Helpers.Wait(3);
            }
            else
            {
                try
                {
                    connection.Open();
                }
                catch
                {
                    Console.WriteLine("Unable to open connection");
                    return;
                }
                PrintPingResponse(connection.Ping());
            }

            // Close connection
            connection.Close();
        }

        private void OpenEvent(Object sender, Ximu3.OpenEventArgs args)
        {
            if (args.result != Ximu3.Result.Ok)
            {
                Console.WriteLine("Unable to open connection");
                return;
            }
            PrintPingResponse(((Ximu3.Connection)sender).Ping());
        }

        private void PrintPingResponse(Ximu3.PingResponse pingResponse)
        {
            if (pingResponse.Interface.Length == 0)
            {
                Console.WriteLine("Ping failed");
                return;
            }
            Console.WriteLine(pingResponse.Interface + ", " + pingResponse.DeviceName + " - " + pingResponse.SerialNumber);
            // Console.WriteLine(pingResponse); // alternative to above
        }
    }
}
